src/Microsoft.ComponentDetection.Detectors/rust/RustSbomDetector.cs

```diff
@@ -3,124 +3,708 @@ namespace Microsoft.ComponentDetection.Detectors.Rust;
+            .OrderBy(p => this.GetDirectoryDepth(p))
+            .ThenBy(p => p, this.pathComparer)
+            .ToList();
+
+        if (tomlPaths.Count > 0)
+        {
+            try
+            {
+                this.Logger.LogInformation("Building Rust ownership map from {Count} Cargo.toml files", tomlPaths.Count);
+                var ownership = await this.metadataContextBuilder.BuildPackageOwnershipMapAsync(tomlPaths, cancellationToken);
+                this.ownershipMap = ownership.PackageToTomls;
+                this.manifestMetadataCache = ownership.ManifestToMetadata;
+                this.Logger.LogInformation(
+                    "Loaded Rust ownership (packages: {PkgCount}) and metadata cache (manifests: {ManifestCount})",
+                    this.ownershipMap?.Count ?? 0,
+                    this.manifestMetadataCache?.Count ?? 0);
+
+                if (ownership.FailedManifests?.Count > 0)
+                {
+                    this.Logger.LogInformation(
+                        "Rust metadata failed for {Count} manifests: {Manifests}",
+                        ownership.FailedManifests.Count,
+                        string.Join(", ", ownership.FailedManifests));
+                }
+            }
+            catch (Exception ex)
+            {
+                this.Logger.LogWarning(ex, "Failed to compute Rust ownership/metadata cache; proceeding without cache");
+                this.ownershipMap = null;
+                this.manifestMetadataCache = null;
+            }
+        }
+        else
+        {
+            this.Logger.LogInformation("No Cargo.toml files found; ownership and metadata cache unavailable");
+            this.ownershipMap = null;
+            this.manifestMetadataCache = null;
+        }
+
+        IEnumerable<ProcessRequest> filteredRequests;
+        if (this.mode == DetectionMode.SBOM_ONLY)
+        {
+            // Only SBOM files, ordered by path ascending
+            filteredRequests = allRequests
+                .Where(r => r.ComponentStream.Location.EndsWith(".cargo-sbom.json", this.pathComparison))
+                .OrderBy(r => r.ComponentStream.Location, this.pathComparer);
+
+            this.Logger.LogInformation("SBOM_ONLY mode: Processing {Count} SBOM files", filteredRequests.Count());
+        }
+        else
+        {
+            // FALLBACK mode: Select Cargo.toml and Cargo.lock files
+            // Order: TOML before LOCK, then depth ascending, then path ascending
+            filteredRequests = allRequests
+                .Where(r =>
+                {
+                    var fileName = Path.GetFileName(r.ComponentStream.Location);
+                    return fileName.Equals("Cargo.toml", this.pathComparison) ||
+                           fileName.Equals("Cargo.lock", this.pathComparison);
+                })
+                .OrderBy(r => Path.GetFileName(r.ComponentStream.Location).Equals("Cargo.lock", this.pathComparison) ? 1 : 0) // TOML before LOCK
```

Comment:
nit: should be create const string for these file names so we can share between its references